12/umbraco-commerce/reference/storefront-api/README.md

```diff
@@ -0,0 +1,611 @@
+
+## Getting Started
+
+The Storefront API works alongside the Content Delivery API so you must first have the [Content Delivery API setup and enabled](https://docs.umbraco.com/umbraco-cms/reference/content-delivery-api#register-the-content-delivery-api-dependencies). With this enabled, you will need to explicitly opt-in to the Storefront API. Below you will find the steps you need to take in order to configure it for your Umbraco project.
+
+### Register the Storefront API dependencies
+1. Open your project's `Startup.cs` file.
+2. Register the API dependencies in the `ConfigureServices` method by adding `.AddStorefrontApi()` inside a `.AddUmbracoCommerce()` call:
+
+{% code title="Startup.cs" %}
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddUmbraco(_env, _config)
+        .AddBackOffice()             
+        .AddWebsite()
+        .AddDeliveryApi()
+        .AddUmbracoCommerce(builder => {
+            builder.AddStorefrontApi();
+        })
+        .AddComposers()
+        .Build();
+}
+```
+{% endcode %}
+
+### Enable the Storefront API
+
+1. Open your project's `appsettings.json`.
+2. Insert the `StorefrontApi` configuration section under `Umbraco:Commerce`.
+3. Add the `Enabled` key and set its value to `true`:
+
+{% code title="appsettings.json" %}
+```json
+{
+    "Umbraco": {
+        "Commerce": {
+            "StorefrontAPi": {
+                "Enabled": true
+            }
+        }
+    }
+}
+```
+{% endcode %}
+
+### Securing the Storefront API
+
+In order to work with the Storefront API many of the endpoints require auhtorization, this authorization is implemented by means of an API Key that must be passed in the header of these requests. The API Key is defined as an additional app setting and can be any value you decide:
+
+{% code title="appsettings.json" %}
+```json
+{
+    "Umbraco": {
+        "Commerce": {
+            "StorefrontAPi": {
+                "Enabled": true,
+                "ApiKey": "ZUynC149dD2N5efs6HN6dCdXlgOVASs6"
+            }
+        }
+    }
+}
+```
+{% endcode %}
+
+## Concepts
+
+Before exploring the API endpoints detailed below, there are a few concepts to keep in mind.
+
+<details>
+
+<summary>Session</summary>
+
+When working with Umbraco Commerce's C# Api, Umbraco Commerce will normally keep track of a series of items for you, such as the current Order ID or the current Language, which it tracks in a cookie. When working in a headless manor however we can no longer rely on cookies for this behaviour and so it becomes the implementors responsibility to remember these items and pass them as Headers to the endpoints that may need to use them for context.
```

Comment:
```suggestion
When working with Umbraco Commerce's C# Api, Umbraco Commerce will normally keep track of a series of items for you. This could be the current Order ID or the current Language, which it tracks in a cookie. When working in a headless manner however we can no longer rely on cookies for this behavior. It becomes the implementor's responsibility to remember these items and pass them as Headers to the endpoints that need to use them for context.
```